### ISSUE TITLE
OSError: [Errno 30] Read-only file system: '/data/stop_addon_state.json'

### ISSUE THREAD

**Issue by whit-tj** [sabeechen/hassio-google-drive-backup#301]
###### Description:
```
I had a notification from Home Assistant alerting me to this error. I searched and found no similar issues in GitHub so wanted to pass it along. I've been using this for about a year, and rarely need to touch it so I haven't been changing any settings, I believe my last update was about a week ago.
```

 Addon version: `0.102.0`
 Home Assistant Version: `0.116.2`
 Supervisor Version: `2020.10.0`
 Supervisor Channel: `stable`
 Hassos Version: `4.15`
 Docker Version: `19.03.11`
 Architecture: `amd64`
 Machine: `qemux86-64`
 Date: `2020-11-04T17:36:36.543984+00:00`
 Timezone: `America/Denver`
 Failure Time: `2020-11-04T17:16:31.318144+00:00`
 Last Good Sync: `2020-11-04T17:16:30.343419+00:00`
 ###### Exception:
 ```
 
  addon/backup/model/coordinator.py:148 (_sync)
  addon/backup/model/model.py:147 (sync)
  addon/backup/model/model.py:181 (createSnapshot)
  addon/backup/ha/hasource.py:181 (create)
  addon/backup/ha/addon_stopper.py:85 (stopAddons)
  addon/backup/ha/addon_stopper.py:137 (_save)
OSError: [Errno 30] Read-only file system: '/data/stop_addon_state.json'

 ```
 Snapshots:
 ```
 {
    "HomeAssistant": {
        "snapshots": 3,
        "retained": 0,
        "deletable": 3,
        "name": "HomeAssistant",
        "latest": "2020-10-29T10:06:50Z",
        "size": "95.9 MB"
    },
    "GoogleDrive": {
        "snapshots": 3,
        "retained": 0,
        "deletable": 3,
        "name": "GoogleDrive",
        "latest": "2020-10-29T10:06:50Z",
        "size": "95.9 MB"
    }
}
 ```
 ###### Config:
 ```
 {
    "Setting.MAX_SNAPSHOTS_IN_HASSIO": 3,
    "Setting.MAX_SNAPSHOTS_IN_GOOGLE_DRIVE": 3,
    "Setting.DAYS_BETWEEN_SNAPSHOTS": 5.0,
    "Setting.SNAPSHOT_TIME_OF_DAY": "03:30",
    "Setting.SEND_ERROR_REPORTS": true,
    "Setting.BACKGROUND_COLOR": "#080808",
    "Setting.ACCENT_COLOR": "#f5a002"
}
 ```
 ###### Addon Logs:
 ```
 11-04 08:16:29 DEBUG [backup.ha.harequests] Making Hassio request: http://hassio/addons/self/info
11-04 08:16:29 DEBUG [backup.ha.harequests] Making Hassio request: http://hassio/info
11-04 08:16:29 DEBUG [backup.ha.harequests] Making Hassio request: http://hassio/core/info
11-04 08:16:29 DEBUG [backup.ha.harequests] Making Hassio request: http://hassio/supervisor/info
11-04 08:16:29 ERROR [backup.model.coordinator] 
  addon/backup/model/coordinator.py:148 (_sync)
  addon/backup/model/model.py:147 (sync)
  addon/backup/model/model.py:181 (createSnapshot)
  addon/backup/ha/hasource.py:181 (create)
  addon/backup/ha/addon_stopper.py:85 (stopAddons)
  addon/backup/ha/addon_stopper.py:137 (_save)
OSError: [Errno 30] Read-only file system: '/data/stop_addon_state.json'

11-04 08:16:29 INFO [backup.model.coordinator] I'll try again in an hour
11-04 09:16:29 DEBUG [backup.model.syncer] Sync requested by Coordinator
11-04 09:16:29 INFO [backup.model.coordinator] Syncing Snapshots
11-04 09:16:29 DEBUG [backup.ha.harequests] Making Hassio request: http://hassio/snapshots
11-04 09:16:29 DEBUG [backup.drive.driverequests] Requesting refreshed Google Drive credentials
11-04 09:16:29 DEBUG [backup.drive.driverequests] Making Google Drive request: https://www.googleapis.com/drive/v3/files/176swUxnCnoZzxS4a7t-DmFVFbkMjC_ey/?fields=id%2Cname%2CappProperties%2Csize%2Ctrashed%2CmimeType%2CmodifiedTime%2Ccapabilities%2Cparents&supportsAllDrives=true
11-04 09:16:30 DEBUG [backup.drive.driverequests] Making Google Drive request: https://www.googleapis.com/drive/v3/files/?q=%27176swUxnCnoZzxS4a7t-DmFVFbkMjC_ey%27+in+parents&fields=nextPageToken%2Cfiles%28id%2Cname%2CappProperties%2Csize%2Ctrashed%2CmimeType%2CmodifiedTime%2Ccapabilities%2Cparents%29&pageSize=100&supportsAllDrives=true&includeItemsFromAllDrives=true&corpora=allDrives
11-04 09:16:30 DEBUG [backup.ha.harequests] Making Hassio request: http://hassio/addons/self/info
11-04 09:16:30 DEBUG [backup.ha.harequests] Making Hassio request: http://hassio/info
11-04 09:16:30 DEBUG [backup.ha.harequests] Making Hassio request: http://hassio/core/info
11-04 09:16:30 DEBUG [backup.ha.harequests] Making Hassio request: http://hassio/supervisor/info
11-04 09:16:30 ERROR [backup.model.coordinator] 
  addon/backup/model/coordinator.py:148 (_sync)
  addon/backup/model/model.py:147 (sync)
  addon/backup/model/model.py:181 (createSnapshot)
  addon/backup/ha/hasource.py:181 (create)
  addon/backup/ha/addon_stopper.py:85 (stopAddons)
  addon/backup/ha/addon_stopper.py:137 (_save)
OSError: [Errno 30] Read-only file system: '/data/stop_addon_state.json'

11-04 09:16:30 INFO [backup.model.coordinator] I'll try again in an hour
11-04 10:16:30 DEBUG [backup.model.syncer] Sync requested by Coordinator
11-04 10:16:30 INFO [backup.model.coordinator] Syncing Snapshots
11-04 10:16:30 DEBUG [backup.ha.harequests] Making Hassio request: http://hassio/snapshots
11-04 10:16:30 DEBUG [backup.drive.driverequests] Requesting refreshed Google Drive credentials
11-04 10:16:30 DEBUG [backup.drive.driverequests] Making Google Drive request: https://www.googleapis.com/drive/v3/files/176swUxnCnoZzxS4a7t-DmFVFbkMjC_ey/?fields=id%2Cname%2CappProperties%2Csize%2Ctrashed%2CmimeType%2CmodifiedTime%2Ccapabilities%2Cparents&supportsAllDrives=true
11-04 10:16:30 DEBUG [backup.drive.driverequests] Making Google Drive request: https://www.googleapis.com/drive/v3/files/?q=%27176swUxnCnoZzxS4a7t-DmFVFbkMjC_ey%27+in+parents&fields=nextPageToken%2Cfiles%28id%2Cname%2CappProperties%2Csize%2Ctrashed%2CmimeType%2CmodifiedTime%2Ccapabilities%2Cparents%29&pageSize=100&supportsAllDrives=true&includeItemsFromAllDrives=true&corpora=allDrives
11-04 10:16:31 DEBUG [backup.ha.harequests] Making Hassio request: http://hassio/addons/self/info
11-04 10:16:31 DEBUG [backup.ha.harequests] Making Hassio request: http://hassio/info
11-04 10:16:31 DEBUG [backup.ha.harequests] Making Hassio request: http://hassio/core/info
11-04 10:16:31 DEBUG [backup.ha.harequests] Making Hassio request: http://hassio/supervisor/info
11-04 10:16:31 ERROR [backup.model.coordinator] 
  addon/backup/model/coordinator.py:148 (_sync)
  addon/backup/model/model.py:147 (sync)
  addon/backup/model/model.py:181 (createSnapshot)
  addon/backup/ha/hasource.py:181 (create)
  addon/backup/ha/addon_stopper.py:85 (stopAddons)
  addon/backup/ha/addon_stopper.py:137 (_save)
OSError: [Errno 30] Read-only file system: '/data/stop_addon_state.json'

11-04 10:16:31 INFO [backup.model.coordinator] I'll try again in an hour
 ```
 ###### Supervisor Logs:
 ```
 [32m20-10-31 04:19:42 INFO (MainThread) [supervisor.homeassistant.api] Updated Home Assistant API token[0m
[32m20-10-31 04:49:19 INFO (MainThread) [supervisor.api.security] /snapshots access from cebe7a76_hassio_google_drive_backup[0m
[32m20-10-31 04:49:44 INFO (MainThread) [supervisor.homeassistant.api] Updated Home Assistant API token[0m
[32m20-10-31 05:19:47 INFO (MainThread) [supervisor.homeassistant.api] Updated Home Assistant API token[0m
[32m20-10-31 05:41:17 INFO (MainThread) [supervisor.host.info] Updating local host information[0m
[32m20-10-31 05:41:17 INFO (MainThread) [supervisor.host.services] Updating service information[0m
[32m20-10-31 05:41:17 INFO (MainThread) [supervisor.host.network] Updating local network information[0m
[32m20-10-31 05:41:17 INFO (MainThread) [supervisor.host.sound] Updating PulseAudio information[0m
Failed to load cookie file from cookie: No such file or directory
[32m20-10-31 05:49:21 INFO (MainThread) [supervisor.api.security] /snapshots access from cebe7a76_hassio_google_drive_backup[0m
[32m20-10-31 05:49:49 INFO (MainThread) [supervisor.homeassistant.api] Updated Home Assistant API token[0m
[32m20-10-31 06:14:26 INFO (MainThread) [supervisor.updater] Fetching update data from https://version.home-assistant.io/stable.json[0m
[32m20-10-31 06:14:40 INFO (MainThread) [supervisor.store.git] Update add-on https://github.com/home-assistant/hassio-addons repository[0m
[32m20-10-31 06:14:40 INFO (MainThread) [supervisor.store.git] Update add-on https://github.com/hassio-addons/repository repository[0m
[32m20-10-31 06:14:40 INFO (MainThread) [supervisor.store.git] Update add-on https://github.com/sabeechen/hassio-google-drive-backup repository[0m
[32m20-10-31 06:14:41 INFO (MainThread) [supervisor.store] Loading add-ons from store: 67 all - 0 new - 0 remove[0m
[32m20-10-31 06:19:51 INFO (MainThread) [supervisor.homeassistant.api] Updated Home Assistant API token[0m
/usr/lib/bashio/jq.sh: line 26: cannot create temp file for here-document: Permission denied
/usr/lib/bashio/jq.sh: line 26: cannot create temp file for here-document: Permission denied
[32m20-10-31 06:49:22 INFO (MainThread) [supervisor.api.security] /snapshots access from cebe7a76_hassio_google_drive_backup[0m
[32m20-10-31 06:49:53 INFO (MainThread) [supervisor.homeassistant.api] Updated Home Assistant API token[0m
/usr/lib/bashio/jq.sh: line 26: cannot create temp file for here-document: Permission denied
/usr/lib/bashio/jq.sh: line 26: cannot create temp file for here-document: Permission denied
/usr/lib/bashio/jq.sh: line 26: cannot create temp file for here-document: Permission denied
/usr/lib/bashio/jq.sh: line 26: cannot create temp file for here-document: Permission denied
/usr/lib/bashio/jq.sh: line 26: cannot create temp file for here-document: Permission denied
/usr/lib/bashio/jq.sh: line 26: cannot create temp file for here-document: Permission denied
/usr/lib/bashio/jq.sh: line 26: cannot create temp file for here-document: Permission denied
/usr/lib/bashio/jq.sh: line 26: cannot create temp file for here-document: Permission denied

 ```
 ###### Home Assistant Core Logs:
 ```
     self.flush()
  File "/usr/local/lib/python3.8/logging/__init__.py", line 1065, in flush
    self.stream.flush()
OSError: [Errno 30] Read-only file system
Call stack:
  File "/usr/local/lib/python3.8/threading.py", line 890, in _bootstrap
    self._bootstrap_inner()
  File "/usr/local/lib/python3.8/threading.py", line 932, in _bootstrap_inner
    self.run()
  File "/usr/local/lib/python3.8/threading.py", line 870, in run
    self._target(*self._args, **self._kwargs)
Message: 'Error in database connectivity during commit: %s. (retrying in %s seconds)'
Arguments: (OperationalError('(sqlite3.OperationalError) attempt to write a readonly database'), 3)
2020-10-31 01:09:40 ERROR (Recorder) [homeassistant.components.recorder] Error executing query: (sqlite3.OperationalError) attempt to write a readonly database
[SQL: INSERT INTO events (event_type, event_data, origin, time_fired, created, context_id, context_user_id, context_parent_id) VALUES (?, ?, ?, ?, ?, ?, ?, ?)]
[parameters: ('state_changed', '{}', 'LOCAL', '2020-10-31 07:09:38.026733', '2020-10-31 07:09:40.056465', '09c2764e1b4811ebaff5b93baa311885', None, None)]
(Background on this error at: http://sqlalche.me/e/13/e3q8)
--- Logging error ---
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/logging/__init__.py", line 1085, in emit
    self.flush()
  File "/usr/local/lib/python3.8/logging/__init__.py", line 1065, in flush
    self.stream.flush()
OSError: [Errno 30] Read-only file system
Call stack:
  File "/usr/local/lib/python3.8/threading.py", line 890, in _bootstrap
    self._bootstrap_inner()
  File "/usr/local/lib/python3.8/threading.py", line 932, in _bootstrap_inner
    self.run()

 ```

**Comment by sabeechen** [comment]
I had another bug about a week ago with a similar error (unable to write to the `/data` directory) that appeared to be a supervisor problem because restarting the host machine made it stop.

That error message in your supervisor logs:
```
/usr/lib/bashio/jq.sh: line 26: cannot create temp file for here-document: Permission denied
```
Is similar to what some users are saying [here](https://community.home-assistant.io/t/home-assistant-not-responding/240710).

I've also seen this issue come up on my personal machine.  

Basically I think there is a supervisor bug.  Maybe more than one.  I'm not sure if its already been fixed or if the supervisor devs are aware of it, but I'd update to the latest version (you're just one version number behind latest release) and restart your host machine to see if that resolves it for now.

**Comment by sabeechen** [comment]
Feel free to reopen the issue if that didn't work for you or there is something else I can help with.
